File: java/src/com/db4o/dg2db4o/chapter7/CasualEmployee.java

package com.db4o.dg2db4o.chapter7;

public class CasualEmployee extends Employee {
    
    double[] _timeRecord;
        final int MAXTIMERECORDS = 10;
        int numberOfTimeRecords;

        public CasualEmployee(String name, String phoneNumber, String email, int employeeNumber, String dob) 
        {
             super(name, phoneNumber, email, employeeNumber, dob);
            _timeRecord = new double[MAXTIMERECORDS];
            numberOfTimeRecords = 0;

        }

        public double[] getTimeRecord()
        {
                return _timeRecord;
        }
        
        public void setTimeRecord(double[] timeRecord)
        {
                _timeRecord = timeRecord;
        }

        public void addTimeRecord(double newRecord)
        {
            if (numberOfTimeRecords < MAXTIMERECORDS)
            {
                _timeRecord[numberOfTimeRecords] = newRecord;
                numberOfTimeRecords++;
            }
        }

        public String toString()
        {
            return super.getName() + " (CasualEmployee)";
        }
    
}

File: csharp/Chapter07/CasualEmployee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace com.db4o.dg2db4o.chapter7
{
    class CasualEmployee : Employee
    {
        double[] _timeRecord;
        const int MAXTIMERECORDS = 10;
        int numberOfTimeRecords;


        public CasualEmployee(string name, string phoneNumber, string email, int employeeNumber, string dob) : base(name, phoneNumber, email, employeeNumber, dob)
        {
            _timeRecord = new double[MAXTIMERECORDS];
            numberOfTimeRecords = 0;

        }

        public double[] TimeRecord
        {
            get
            {
                return _timeRecord;
            }
            set
            {
                _timeRecord = value;
            }
        }

        public void addTimeRecord(double newRecord)
        {
            if (numberOfTimeRecords < MAXTIMERECORDS)
            {
                _timeRecord[numberOfTimeRecords] = newRecord;
                numberOfTimeRecords++;
            }
        }

        public override string ToString()
        {
            return base.Name + " (CasualEmployee)";
        }
    }
}
